bson/codec_options.py

```diff
@@ -140,8 +140,8 @@ def __init__(
     ) -> None:
         self.__type_codecs = list(type_codecs or [])
         self._fallback_encoder = fallback_encoder
-        self._encoder_map: Dict[Any, Any] = {}
-        self._decoder_map: Dict[Any, Any] = {}
+        self._encoder_map: dict[Any, Any] = {}
```

Comment:
This should be kept as `Dict` to be consistent with the `pymongo` package typing.

Reply:
Those will be changed as part of [PYTHON-3905](https://jira.mongodb.org/browse/PYTHON-3905)